src/execution/operator/persistent/physical_copy_to_file.cpp

```diff
@@ -211,6 +216,10 @@ unique_ptr<GlobalFunctionData> PhysicalCopyToFile::CreateFileState(ClientContext
 	idx_t this_file_offset = g.last_file_offset++;
 	auto &fs = FileSystem::GetFileSystem(context);
 	string output_path(filename_pattern.CreateFilename(fs, file_path, file_extension, this_file_offset));
+	if (return_files) {
+		auto l = g.lock.GetExclusiveLock();
+		g.file_names.emplace_back(output_path);
```

Comment:
Can we move this into a function in the global state (`AddFileName`, that takes as parameter the lock) so that we cannot accidentally call it without having the lock?